rust/main/submitter/src/payload/types.rs

```diff
@@ -78,4 +78,8 @@ impl FullPayload {
     pub fn set_status(&mut self, status: PayloadStatus) {
         self.status = status;
     }
+
+    pub fn details(&self) -> PayloadDetails {
+        self.details.clone()
+    }
```

Comment:
I wonder if we want user of the method to decide if they want to clone the instance?

Reply:
good idea